pkg/agent/plugin/workloadattestor/docker/docker.go

```diff
@@ -0,0 +1,154 @@
+}
+
+type dockerPluginConfig struct {
+	// DockerSocketPath is the location of the docker daemon socket (default: "unix:///var/run/docker.sock" on unix).
+	DockerSocketPath string `hcl:"docker_socket_path"`
+	// DockerVersion is the API version of the docker daemon (default: "1.25").
+	DockerVersion string `hcl:"docker_version"`
+	// CgroupPrefix is the cgroup prefix to look for in the cgroup entries (e.g. "/docker").
+	CgroupPrefix string `hcl:"cgroup_prefix"`
+	// CgroupContainerIndex is the index within the cgroup path where the container ID should be found (default: 0).
+	CgroupContainerIndex int `hcl:"cgroup_container_index"`
+}
+
+func (p *dockerPlugin) Attest(ctx context.Context, req *workloadattestor.AttestRequest) (*workloadattestor.AttestResponse, error) {
+	p.mtx.RLock()
+	defer p.mtx.RUnlock()
+
+	cgroupList, err := cgroups.GetCgroups(req.Pid, p.fs)
+	if err != nil {
+		return nil, err
+	}
+
+	var containerID string
+	var hasDockerEntries bool
+	for _, cgroup := range cgroupList {
+		// We are only interested in cgroup entries that match our desired prefix. Example entry:
+		// "10:perf_event:/docker/2235ebefd9babe0dde4df4e7c49708e24fb31fb851edea55c0ee29a18273cdf4"
+		if !strings.HasPrefix(cgroup.GroupPath, p.cgroupPrefix) {
+			continue
+		}
+		hasDockerEntries = true
+
+		parts := strings.Split(cgroup.GroupPath, "/")
+
+		if len(parts) <= p.cgroupContainerIndex+1 {
+			log.Printf("docker entry found, but without container id: %v", cgroup.GroupPath)
+			continue
+		}
+		containerID = parts[p.cgroupContainerIndex+1]
+		break
+	}
+
+	if !hasDockerEntries {
+		return nil, fmt.Errorf("workloadattestor/docker: no cgroup entries found with prefix %q", p.cgroupPrefix)
+	}
+	if containerID == "" {
+		return nil, fmt.Errorf("workloadattestor/docker: no cgroup %q entries found at index %d", p.cgroupPrefix, p.cgroupContainerIndex)
+	}
+
+	container, err := p.docker.ContainerInspect(ctx, containerID)
+	if err != nil {
+		return nil, err
+	}
+
+	return &workloadattestor.AttestResponse{
+		Selectors: getSelectorsFromConfig(container.Config),
+	}, nil
+}
+
+func getSelectorsFromConfig(cfg *container.Config) []*common.Selector {
+	var selectors []*common.Selector
+	for label, value := range cfg.Labels {
+		selectors = append(selectors, &common.Selector{
+			Type:  _selectorType,
+			Value: fmt.Sprintf("%s:%s:%s", _subselectorLabel, label, value),
+		})
+	}
+	if cfg.Image != "" {
+		selectors = append(selectors, &common.Selector{
+			Type:  _selectorType,
+			Value: fmt.Sprintf("%s:%s", _subselectorImageID, cfg.Image),
+		})
+	}
+	return selectors
+}
+
+func (p *dockerPlugin) Configure(ctx context.Context, req *spi.ConfigureRequest) (*spi.ConfigureResponse, error) {
+	p.mtx.Lock()
+	defer p.mtx.Unlock()
+
+	// Parse HCL config payload into config struct
+	config := &dockerPluginConfig{}
+	hclTree, err := hcl.Parse(req.Configuration)
+	if err != nil {
+		return &spi.ConfigureResponse{ErrorList: []string{err.Error()}}, err
```

Comment:
i know this was likely a pattern followed from other plugins, but we should just return `nil, err` here. The ErrorList stuff isn't used and gRPC ignores the returned response anyway if an error is returned. I've been meaning to clean up the plugins.... :)